tokio/tests/rt_metrics.rs

```diff
@@ -86,16 +86,50 @@ fn active_tasks_count() {
     let rt = current_thread();
     let metrics = rt.metrics();
     assert_eq!(0, metrics.active_tasks_count());
-    rt.spawn(async move {
+    rt.block_on(rt.spawn(async move {
         assert_eq!(1, metrics.active_tasks_count());
-    });
+    }))
+    .unwrap();
 
     let rt = threaded();
     let metrics = rt.metrics();
     assert_eq!(0, metrics.active_tasks_count());
-    rt.spawn(async move {
+    rt.block_on(rt.spawn(async move {
         assert_eq!(1, metrics.active_tasks_count());
-    });
+    }))
+    .unwrap();
+}
+
+#[test]
+fn spawned_tasks_count() {
+    let rt = current_thread();
+    let metrics = rt.metrics();
+    assert_eq!(0, metrics.spawned_tasks_count());
+
+    rt.block_on(rt.spawn(async move {
+        assert_eq!(1, metrics.spawned_tasks_count());
+    }))
+    .unwrap();
+
+    assert_eq!(1, rt.metrics().spawned_tasks_count());
+
+    let rt = threaded();
+    let metrics = rt.metrics();
+    assert_eq!(0, metrics.spawned_tasks_count());
+
+    rt.block_on(rt.spawn(async move {
+        assert_eq!(1, metrics.spawned_tasks_count());
+    }))
+    .unwrap();
+
+    for _ in 0..100 {
+        if rt.metrics().spawned_tasks_count() == 1 {
+            return;
+        }
+        // on single threaded machines (like in CI), we need to force the OS to run the runtime threads
+        std::thread::sleep(std::time::Duration::from_millis(1));
+    }
```

Comment:
Is this loop really necessary? Your use of `block_on` already causes you to sleep until the task has been spawned, so surely the counter is guaranteed to be correct?

Reply:
maybe not anymore. before the test also checked that the stopped task count dropped, and on some of the test platforms that seemed flaky (loop was added as per your suggestion: https://github.com/tokio-rs/tokio/pull/6114#discussion_r1382584192)

Now that the test only checks start count, it should be fine.

Reply:
Yeah. I just added an assert_eq(0) to the active count test and that is failing in some cases. This is what the retry was for. I will remove this assert as its not relevant for this PR